lamindb/models/artifact.py

```diff
@@ -523,45 +523,53 @@ def log_storage_hint(
     logger.hint(hint)
 
 
-def data_is_anndata(data: AnnData | UPathStr) -> bool:
-    if isinstance(data, AnnData):
+def data_is_scversedatastructure(
+    data: ScverseDataStructures | UPathStr,
+    class_name: Literal["AnnData", "MuData", "SpatialData"],
+) -> bool:
+    file_suffix = None
+    if class_name == "AnnData":
+        file_suffix = ".h5ad"
+    elif class_name == "MuData":
+        file_suffix = ".h5mu"
+    # SpatialData does not have a unique suffix but `.zarr`
+
+    if hasattr(data, "__class__") and data.__class__.__name__ == class_name:
         return True
+
+    data_type = class_name.lower()
     if isinstance(data, (str, Path, UPath)):
         data_path = UPath(data)
-        if ".h5ad" in data_path.suffixes:  # ".h5ad.gz" is a valid suffix
+
+        if file_suffix and f"{file_suffix}" in data_path.suffixes:
             return True
-        elif data_path.suffix == ".zarr":
-            # ".anndata.zarr" is a valid suffix (core.storage._valid_suffixes)
-            # TODO: the suffix based check should likely be moved to identify_zarr_type
-            if ".anndata" in data_path.suffixes:
+
+        if data_path.suffix == ".zarr":
+            type_suffix = f".{data_type}"
+            if type_suffix in data_path.suffixes:
                 return True
+
             # check only for local, expensive for cloud
             if fsspec.utils.get_protocol(data_path.as_posix()) == "file":
-                return identify_zarr_type(data_path) == "anndata"
+                return (
+                    identify_zarr_type(
+                        data_path if class_name == "AnnData" else data,
+                        check=True if class_name == "AnnData" else False,
```

Comment:
I think a False for all 3 is also okay but I kept the behavior as it was for now.